source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -363,9 +368,12 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
         {
             format = decor->getFormat();
         }
+        // If the texture has no format decoration, try to infer it from the type.
         if (format == ImageFormat::unknown)
         {
-            // If the texture has no format decoration, try to infer it from the type.
+            // Do not infer format when shape is a subpass because SPRI-V spec defines
+            // subpass to have an unknown format 
+            if (textureType->GetBaseShape() == SLANG_TEXTURE_SUBPASS) return;
```

Comment:
If I understood the comment correctly, you could separate the subpass logic from the existing logic like this:
```
        if (textureType->GetBaseShape() == SLANG_TEXTURE_SUBPASS)
        {
            // Do not infer format when shape is a subpass because SPRI-V spec defines
            // subpass to have an unknown format 
            SLANG_ASSERT(format == ImageFormat::unknown);
            return;
        }

        if (format == ImageFormat::unknown)
        {
            // If the texture has no format decoration, try to infer it from the type.
```

Reply:
I agree. I should also add a diagnostic error to warn users that the format should be 'unknown'

I will add this to tests.